aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2175,7 +2175,14 @@ std::vector<Tensor> meshgrid(TensorList tensors) {
 // Numpy-style `a.T`: returns the tensor
 // with dims reversed
 Tensor numpy_T(const Tensor &self) {
-  int64_t n = self.dim();
+  const auto n = self.dim();
+  if (n != 2 && n != 0) {
+    TORCH_WARN_ONCE(
+        "The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated ",
+        "and it will throw an error in a future release. Consider `x.mT` to transpose batches of matrices",
+        "or `x.permute(*torch.arange(x.ndim - 1, -1, -1))` to reverse the dimensions of a tensor."
```

Comment:
I would if this functionality is so common that we should add a special operator, like `reverse_dims` as a drop-in replacement?

Reply:
I do not know of any use case of this. I just wrote how to implement the functionality in PyTorch in case anyone wants to use it, but I highly doubt it.

Reply:
I think you're right. In that case let's put the warning in and not worry about adding a new op. We can see how frequent this warning is or if the community complains.